### ISSUE TITLE
None of the effects work on Fedora 28, 29

### ISSUE THREAD

**Issue by hpg-mheck** [wwmm/pulseeffects#482]
Once all dependencies have been fully installed, the system does build on Fedora 28 (it will not build on Fedora 29 due to Meson version issues), but none of the effects do anything.

**Comment by hpg-mheck** [comment]
The system will recognize the combination of the effects system on the selected output device as a valid default output device, and the basic "Front Left", "Front Right" tests work.  Additionally, the master level sliders seem to work, and the toggles to bypass PulseEffects entirely for individual applications result in a large amplitude change (regardless of slider at max).  But none of the individual effects seem to work, regardless of their supporting libraries.

**Comment by hpg-mheck** [comment]
What?  Closed?  I didn't close this...

**Comment by hpg-mheck** [comment]
The system will recognize the combination of the effects system on the selected output device as a valid default output device, and the basic "Front Left", "Front Right" tests work. Additionally, the master level sliders seem to work, and the toggles to bypass PulseEffects entirely for individual applications result in a large amplitude change (regardless of slider at max). But none of the individual effects seem to work, regardless of their supporting libraries.

**Comment by wwmm** [comment]
It seems that the plugins are not installed or that GStreamer can not find them for some reason. Run in debug mode `G_MESSAGES_DEBUG=pulseeffects pulseeffects` and put here what is printed in the terminal.

**Comment by hpg-mheck** [comment]
Thank you.  I'll provide that output next time I'm at that site, should be some time this week.  We can also see if the usual round of Fedora updates has had any effect...

**Comment by rhoot** [comment]
Same thing in Fedora 30, the application runs but none of the controls actually do anything. When running in debug mode as suggested there's a bunch of "plugin was not found!" lines (see the attached file). I tried installing both "good" and "bad" gstreamer plugins to no avail.

```
$ ll /usr/lib64/gstreamer-1.0/libgstequalizer.so 
-rwxr-xr-x. 1 root root 42632 Apr 23 11:21 /usr/lib64/gstreamer-1.0/libgstequalizer.so
```

This is with PulseEffects 4.5.0.

[pulseeffects-debug.txt](https://github.com/wwmm/pulseeffects/files/3230700/pulseeffects-debug.txt)

**Comment by wwmm** [comment]
We do not use GStreamer equalizer anymore. We switched to the one from Linux Studio Plugins https://lsp-plug.in/?page=manuals&section=para_equalizer_x32_lr. At least on Arch Linux the package is named `lsp-plugins`. When it is installed the command `gst-inspect-1.0 | grep -i lsp | grep -i lv2` will print lots of related plugins.

Assuming you have already installed all the optional dependencies something else you may try is deleting GStreamer cache `rm -rf ~/.cache/gstreamer-1.0`

**Comment by wwmm** [comment]
Our README file has a dependency list

**Comment by roberttucker** [comment]
How exactly does one install the LSP plugins for use for PE? I think I've succeeded with LADSPA but not with VST.
<details>
<summary> gst-inspect </summary>
`gst-inspect-1.0 | grep -i lsp
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-phase-detector: Phase Detector - Phasendetektor
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-comp-delay-mono: no LADSPA description available
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-comp-delay-stereo: no LADSPA description available
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-comp-delay-x2-stereo: no LADSPA description available
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-spectrum-analyzer-x1: Spectrum Analyzer x1 - Spektrumanalysator x1
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-spectrum-analyzer-x2: Spectrum Analyzer x2 - Spektrumanalysator x2
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-spectrum-analyzer-x4: Spectrum Analyzer x4 - Spektrumanalysator x4
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-spectrum-analyzer-x8: Spectrum Analyzer x8 - Spektrumanalysator x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-spectrum-analyzer-x12: Spectrum Analyzer x12 - Spektrumanalysator x12
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-spectrum-analyzer-x16: Spectrum Analyzer x16 - Spektrumanalysator x16
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x16-mono: Parametric Equalizer x16 Mono - Parametrischer Entzerrer x16 Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x32-mono: Parametric Equalizer x32 Mono - Parametrischer Entzerrer x32 Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x16-stereo: Parametric Equalizer x16 Stereo - Parametrischer Entzerrer x16 Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x32-stereo: Parametric Equalizer x32 Stereo - Parametrischer Entzerrer x32 Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x16-lr: Parametric Equalizer x16 LeftRight - Parametrischer Entzerrer x16 LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x32-lr: Parametric Equalizer x32 LeftRight - Parametrischer Entzerrer x32 LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x16-ms: Parametric Equalizer x16 MidSide - Parametrischer Entzerrer x16 MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-para-equalizer-x32-ms: Parametric Equalizer x32 MidSide - Parametrischer Entzerrer x32 MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x16-mono: Graphic Equalizer x16 Mono - Grafischer Entzerrer x16 Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x16-stereo: Graphic Equalizer x16 Stereo - Grafischer Entzerrer x16 Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x16-lr: Graphic Equalizer x16 LeftRight - Grafischer Entzerrer x16 LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x16-ms: Graphic Equalizer x16 MidSide - Grafischer Entzerrer x16 MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x32-mono: Graphic Equalizer x32 Mono - Grafischer Entzerrer x32 Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x32-stereo: Graphic Equalizer x32 Stereo - Grafischer Entzerrer x32 Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x32-lr: Graphic Equalizer x32 LeftRight - Grafischer Entzerrer x32 LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-graph-equalizer-x32-ms: Graphic Equalizer x32 MidSide - Grafischer Entzerrer x32 MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-compressor-mono: Compressor Mono - Kompressor Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-compressor-stereo: Compressor Stereo - Kompressor Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-compressor-lr: Compressor LeftRight - Kompressor LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-compressor-ms: Compressor MidSide - Kompressor MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-compressor-mono: Sidechain Compressor Mono - Sidechain-Kompressor Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-compressor-stereo: Sidechain Compressor Stereo - Sidechain-Kompressor Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-compressor-lr: Sidechain Compressor LeftRight - Sidechain-Kompressor LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-compressor-ms: Sidechain Compressor MidSide - Sidechain-Kompressor MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-dyna-processor-mono: Dynamic Processor Mono - Dynamikprozessor Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-dyna-processor-stereo: Dynamic Processor Stereo - Dynamikprozessor Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-dyna-processor-lr: Dynamic Processor LeftRight - Dynamikprozessor LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-dyna-processor-ms: Dynamic Processor MidSide - Dynamikprozessor MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-dyna-processor-mono: Sidechain Dynamic Processor Mono - Sidechain-Dynamikprozessor Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-dyna-processor-stereo: Sidechain Dynamic Processor Stereo - Sidechain-Dynamikprozessor Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-dyna-processor-lr: Sidechain Dynamic Processor LeftRight - Sidechain-Dynamikprozessor LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-dyna-processor-ms: Sidechain Dynamic Processor MidSide - Sidechain-Dynamikprozessor MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-expander-mono: Expander Mono - Expander Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-expander-stereo: Expander Stereo - Expander Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-expander-lr: Expander LeftRight - Expander LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-expander-ms: Expander MidSide - Expander MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-expander-mono: Sidechain Expander Mono - Sidechain-Expander Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-expander-stereo: Sidechain Expander Stereo - Sidechain-Expander Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-expander-lr: Sidechain Expander LeftRight - Sidechain-Expander LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-expander-ms: Sidechain Expander MidSide - Sidechain-Expander MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-gate-mono: Gate Mono - Gate Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-gate-stereo: Gate Stereo - Gate Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-gate-lr: Gate LeftRight - Gate LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-gate-ms: Gate MidSide - Gate MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-gate-mono: Sidechain Gate Mono - Sidechain-Gate Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-gate-stereo: Sidechain Gate Stereo - Sidechain-Gate Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-gate-lr: Sidechain Gate LeftRight - Sidechain-Gate LeftRight
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-gate-ms: Sidechain Gate MidSide - Sidechain-Gate MidSide
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-limiter-mono: Limiter Mono - Begrenzer Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-limiter-stereo: Limiter Stereo - Begrenzer Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-limiter-mono: Sidechain Limiter Mono - Sidechain-Begrenzer Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-limiter-stereo: Sidechain Limiter Stereo - Sidechain-Begrenzer Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-slap-delay-mono: Slapback Delay Mono - Slapback-Delay Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-slap-delay-stereo: Slapback Delay Stereo - Slapback-Delay Stereo
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-oscillator-mono: Oscillator Mono - Oszillator Mono
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-latency-meter: no LADSPA description available
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-mb-compressor-mono: Multiband Compressor Mono x8 - Multi-band Kompressor Mono x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-mb-compressor-stereo: Multiband Compressor Stereo x8 - Multi-band Kompressor Stereo x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-mb-compressor-lr: Multiband Compressor LeftRight x8 - Multi-band Kompressor LeftRight x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-mb-compressor-ms: Multiband Compressor MidSide x8 - Multi-band Kompressor MidSide x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-mb-compressor-mono: Sidechain Multiband Compressor Mono x8 - Sidechain Multi-band Kompressor Mono x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-mb-compressor-stereo: Sidechain Multiband Compressor Stereo x8 - Sidechain Multi-band Kompressor Stereo x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-mb-compressor-lr: Sidechain Multiband Compressor LeftRight x8 - Sidechain Multi-band Kompressor LeftRight x8
ladspa:  ladspa-lsp-plugins-ladspa-so-http---lsp-plug-in-plugins-ladspa-sc-mb-compressor-ms: Sidechain Multiband Compressor MidSide x8 - Sidechain Multi-band Kompressor MidSide x8`
</details>

$ echo $VST_PATH
/usr/lib64/vst:

<details>
<summary>VST files </summary>
$ ls /usr/lib64/vst/lsp-plugins-lxvst-1.1.9
lsp-plugins-vst-comp-delay-mono.so
lsp-plugins-vst-comp-delay-stereo.so
lsp-plugins-vst-comp-delay-x2-stereo.so
lsp-plugins-vst-compressor-lr.so
lsp-plugins-vst-compressor-mono.so
lsp-plugins-vst-compressor-ms.so
lsp-plugins-vst-compressor-stereo.so
lsp-plugins-vst-core-1.1.9.so
lsp-plugins-vst-dyna-processor-lr.so
lsp-plugins-vst-dyna-processor-mono.so
lsp-plugins-vst-dyna-processor-ms.so
lsp-plugins-vst-dyna-processor-stereo.so
lsp-plugins-vst-expander-lr.so
lsp-plugins-vst-expander-mono.so
lsp-plugins-vst-expander-ms.so
lsp-plugins-vst-expander-stereo.so
lsp-plugins-vst-gate-lr.so
lsp-plugins-vst-gate-mono.so
lsp-plugins-vst-gate-ms.so
lsp-plugins-vst-gate-stereo.so
lsp-plugins-vst-graph-equalizer-x16-lr.so
lsp-plugins-vst-graph-equalizer-x16-mono.so
lsp-plugins-vst-graph-equalizer-x16-ms.so
lsp-plugins-vst-graph-equalizer-x16-stereo.so
lsp-plugins-vst-graph-equalizer-x32-lr.so
lsp-plugins-vst-graph-equalizer-x32-mono.so
lsp-plugins-vst-graph-equalizer-x32-ms.so
lsp-plugins-vst-graph-equalizer-x32-stereo.so
lsp-plugins-vst-impulse-responses-mono.so
lsp-plugins-vst-impulse-responses-stereo.so
lsp-plugins-vst-impulse-reverb-mono.so
lsp-plugins-vst-impulse-reverb-stereo.so
lsp-plugins-vst-latency-meter.so
lsp-plugins-vst-limiter-mono.so
lsp-plugins-vst-limiter-stereo.so
lsp-plugins-vst-mb-compressor-lr.so
lsp-plugins-vst-mb-compressor-mono.so
lsp-plugins-vst-mb-compressor-ms.so
lsp-plugins-vst-mb-compressor-stereo.so
lsp-plugins-vst-multisampler-x12-do.so
lsp-plugins-vst-multisampler-x12.so
lsp-plugins-vst-multisampler-x24-do.so
lsp-plugins-vst-multisampler-x24.so
lsp-plugins-vst-multisampler-x48-do.so
lsp-plugins-vst-multisampler-x48.so
lsp-plugins-vst-oscillator-mono.so
lsp-plugins-vst-para-equalizer-x16-lr.so
lsp-plugins-vst-para-equalizer-x16-mono.so
lsp-plugins-vst-para-equalizer-x16-ms.so
lsp-plugins-vst-para-equalizer-x16-stereo.so
lsp-plugins-vst-para-equalizer-x32-lr.so
lsp-plugins-vst-para-equalizer-x32-mono.so
lsp-plugins-vst-para-equalizer-x32-ms.so
lsp-plugins-vst-para-equalizer-x32-stereo.so
lsp-plugins-vst-phase-detector.so
lsp-plugins-vst-profiler-mono.so
lsp-plugins-vst-profiler-stereo.so
lsp-plugins-vst-sampler-mono.so
lsp-plugins-vst-sampler-stereo.so
lsp-plugins-vst-sc-compressor-lr.so
lsp-plugins-vst-sc-compressor-mono.so
lsp-plugins-vst-sc-compressor-ms.so
lsp-plugins-vst-sc-compressor-stereo.so
lsp-plugins-vst-sc-dyna-processor-lr.so
lsp-plugins-vst-sc-dyna-processor-mono.so
lsp-plugins-vst-sc-dyna-processor-ms.so
lsp-plugins-vst-sc-dyna-processor-stereo.so
lsp-plugins-vst-sc-expander-lr.so
lsp-plugins-vst-sc-expander-mono.so
lsp-plugins-vst-sc-expander-ms.so
lsp-plugins-vst-sc-expander-stereo.so
lsp-plugins-vst-sc-gate-lr.so
lsp-plugins-vst-sc-gate-mono.so
lsp-plugins-vst-sc-gate-ms.so
lsp-plugins-vst-sc-gate-stereo.so
lsp-plugins-vst-sc-limiter-mono.so
lsp-plugins-vst-sc-limiter-stereo.so
lsp-plugins-vst-sc-mb-compressor-lr.so
lsp-plugins-vst-sc-mb-compressor-mono.so
lsp-plugins-vst-sc-mb-compressor-ms.so
lsp-plugins-vst-sc-mb-compressor-stereo.so
lsp-plugins-vst-slap-delay-mono.so
lsp-plugins-vst-slap-delay-stereo.so
lsp-plugins-vst-spectrum-analyzer-x12.so
lsp-plugins-vst-spectrum-analyzer-x16.so
lsp-plugins-vst-spectrum-analyzer-x1.so
lsp-plugins-vst-spectrum-analyzer-x2.so
lsp-plugins-vst-spectrum-analyzer-x4.so
lsp-plugins-vst-spectrum-analyzer-x8.so
lsp-plugins-vst-trigger-midi-mono.so
lsp-plugins-vst-trigger-midi-stereo.so
lsp-plugins-vst-trigger-mono.so
lsp-plugins-vst-trigger-stereo.so
</details>
<details>
<summary> DEBUG </summary>
G_MESSAGES_DEBUG=pulseeffects pulseeffects
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.911: application: PE version: 4.5.0
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.913: pulse_manager: context is connecting
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.913: pulse_manager: context is authorizing
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.914: pulse_manager: context is setting name
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.914: pulse_manager: context is ready
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.914: pulse_manager: connected to: unix:/run/user/1000/pulse/native
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.914: pulse_manager: protocol version: 32
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.915: pulse_manager: Pulseaudio version: 12.2-rebootstrapped
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.915: pulse_manager: default pulseaudio source: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.915: pulse_manager: default pulseaudio sink: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.915: pulse_manager: loading Pulseeffects applications output sink...
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.915: pulse_manager: default pulseaudio sink sampling rate: 44100 Hz
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.915: pulse_manager: default pulseaudio sink audio format: s16le
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.916: pulse_manager: sink loaded
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.916: pulse_manager: loading Pulseeffects microphone output sink...
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.916: pulse_manager: default pulseaudio source sampling rate: 44100 Hz
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.916: pulse_manager: default pulseaudio source audio format: s16le
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.917: pulse_manager: sink loaded
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.937: sie: using input device: PulseEffects_apps.monitor
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.937: sie: using output device: alsa_output.pci-0000_00_1b.0.analog-stereo

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.938: limiter plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.938: compressor plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.938: filter plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.938: equalizer plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.938: reverb plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.938: bass_enhancer plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.939: exciter plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.939: crossfeed plugin was not found!
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.941: sie: maximizer is already disabled

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.941: multiband_compressor plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.941: loudness plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.942: gate plugin was not found!
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.945: sie: pitch is already disabled

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.945: multiband_gate plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.945: deesser plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.945: stereo_tools plugin was not found!
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.949: sie: convolver is already disabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.952: sie: crystalizer is already disabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.953: sie: autogain is already disabled

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.954: delay plugin was not found!
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.960: soe: using output device: PulseEffects_mic
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.960: soe: using input device: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.960: limiter plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.960: compressor plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.961: filter plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.961: equalizer plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.961: reverb plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.961: gate plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.961: deesser plugin was not found!
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.963: soe: pitch is already disabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.969: soe: webrtc is already disabled

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.969: multiband_compressor plugin was not found!

(pulseeffects:5663): pulseeffects-WARNING **: 01:56:31.969: multiband_gate plugin was not found!
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.971: presets_manager: user presets directory already exists: /home/rt/.config/PulseEffects
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.971: presets_manager: user presets directory already exists: /home/rt/.config/PulseEffects/input
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.971: presets_manager: user presets directory already exists: /home/rt/.config/PulseEffects/output
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:31.971: presets_manager: user presets directory already exists: /home/rt/.config/PulseEffects/autoload
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:32.323: convolver_ui: irs directory already exists: /home/rt/.config/PulseEffects/irs
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:32.664: sie: spectrum enabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:32.664: soe: spectrum enabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:32.835: sie: convolver is already disabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:32.835: soe: pitch is already disabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:32.835: soe: webrtc is already disabled
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: added sink: alsa_output.pci-0000_01_00.1.hdmi-stereo-extra2
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: sie: using output device: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: output device changed: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: added sink: alsa_output.pci-0000_00_1b.0.analog-stereo
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: added source: alsa_output.pci-0000_01_00.1.hdmi-stereo-extra2.monitor
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: soe: using input device: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: input device changed: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: added source: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
(pulseeffects:5663): pulseeffects-DEBUG: 01:56:33.100: pulse_settings_ui: added source: alsa_input.pci-0000_00_1b.0.analog-stereo
</details>

**Comment by wwmm** [comment]
We do not use VST. You have to install the LV2 libraries for LSP to work with PE.

**Comment by wwmm** [comment]
The same is true for Calf plugins. You need the its LV2 libraries.

**Comment by roberttucker** [comment]
I thought I'd added the LV2 library:

ls /usr/lib64/lv2/lsp-plugins.lv2
comp_delay_mono.ttl        gate_mono.ttl                   limiter_mono.ttl           para_equalizer_x16_mono.ttl    sc_dyna_processor_lr.ttl      sc_mb_compressor_ms.ttl
comp_delay_stereo.ttl      gate_ms.ttl                     limiter_stereo.ttl         para_equalizer_x16_ms.ttl      sc_dyna_processor_mono.ttl    sc_mb_compressor_stereo.ttl
comp_delay_x2_stereo.ttl   gate_stereo.ttl                 lsp-plugins-lv2.so         para_equalizer_x16_stereo.ttl  sc_dyna_processor_ms.ttl      slap_delay_mono.ttl
compressor_lr.ttl          graph_equalizer_x16_lr.ttl      manifest.ttl               para_equalizer_x32_lr.ttl      sc_dyna_processor_stereo.ttl  slap_delay_stereo.ttl
compressor_mono.ttl        graph_equalizer_x16_mono.ttl    mb_compressor_lr.ttl       para_equalizer_x32_mono.ttl    sc_expander_lr.ttl            spectrum_analyzer_x12.ttl
compressor_ms.ttl          graph_equalizer_x16_ms.ttl      mb_compressor_mono.ttl     para_equalizer_x32_ms.ttl      sc_expander_mono.ttl          spectrum_analyzer_x16.ttl
compressor_stereo.ttl      graph_equalizer_x16_stereo.ttl  mb_compressor_ms.ttl       para_equalizer_x32_stereo.ttl  sc_expander_ms.ttl            spectrum_analyzer_x1.ttl
dyna_processor_lr.ttl      graph_equalizer_x32_lr.ttl      mb_compressor_stereo.ttl   phase_detector.ttl             sc_expander_stereo.ttl        spectrum_analyzer_x2.ttl
dyna_processor_mono.ttl    graph_equalizer_x32_mono.ttl    multisampler_x12_do.ttl    profiler_mono.ttl              sc_gate_lr.ttl                spectrum_analyzer_x4.ttl
dyna_processor_ms.ttl      graph_equalizer_x32_ms.ttl      multisampler_x12.ttl       profiler_stereo.ttl            sc_gate_mono.ttl              spectrum_analyzer_x8.ttl
dyna_processor_stereo.ttl  graph_equalizer_x32_stereo.ttl  multisampler_x24_do.ttl    sampler_mono.ttl               sc_gate_ms.ttl                trigger_midi_mono.ttl
expander_lr.ttl            impulse_responses_mono.ttl      multisampler_x24.ttl       sampler_stereo.ttl             sc_gate_stereo.ttl            trigger_midi_stereo.ttl
expander_mono.ttl          impulse_responses_stereo.ttl    multisampler_x48_do.ttl    sc_compressor_lr.ttl           sc_limiter_mono.ttl           trigger_mono.ttl
expander_ms.ttl            impulse_reverb_mono.ttl         multisampler_x48.ttl       sc_compressor_mono.ttl         sc_limiter_stereo.ttl         trigger_stereo.ttl
expander_stereo.ttl        impulse_reverb_stereo.ttl       oscillator_mono.ttl        sc_compressor_ms.ttl           sc_mb_compressor_lr.ttl
gate_lr.ttl                latency_meter.ttl               para_equalizer_x16_lr.ttl  sc_compressor_stereo.ttl       sc_mb_compressor_mono.ttl

**Comment by wwmm** [comment]
Try to remove GStreamer cache `rm -rf ~/.cache/gstreamer-1.0`. When `gst-inspect` can't find the plugins a corrupted or outdated cache is usually the source of the problem

**Comment by roberttucker** [comment]
Still the same DEBUG report. Does lsp-plugins-profile need to be installed?

**Comment by wwmm** [comment]
I don't think so. Specially because Calf plugins are also not being found. In case it helps this is how Arch Linux is building LSP https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/lsp-plugins. No issues here. But I still think the problem is somewhere else. Is the library `lilv` installed? GStreamer needs it to find LV2 plugins.

**Comment by wwmm** [comment]
Now I remember that just installing `lilv` is not enough. `gst-plugins-bad` has to be compiled with it. IF Fedora maintainers disabled lilv at compile time LV2 plugins are never going to be found.

**Comment by wwmm** [comment]
This is how Arch Linux compiles `gst-plugins-bad` https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/gst-plugins-bad

**Comment by Vascom** [comment]
Now it is built with lilv https://bodhi.fedoraproject.org/updates/FEDORA-2019-48e0e52af7
Most of plugins now visible for pulseeffects but not all.
```
(pulseeffects:11759): pulseeffects-WARNING **: 09:09:11.579: compressor plugin was not found!
(pulseeffects:11759): pulseeffects-WARNING **: 09:09:11.580: equalizer plugin was not found!
(pulseeffects:11759): pulseeffects-WARNING **: 09:09:11.586: crossfeed plugin was not found!
(pulseeffects:11759): pulseeffects-WARNING **: 09:09:11.614: delay plugin was not found!
(pulseeffects:11759): pulseeffects-WARNING **: 09:09:11.621: compressor plugin was not found!
(pulseeffects:11759): pulseeffects-WARNING **: 09:09:11.622: equalizer plugin was not found!
```
Please help me debug why this plugins not catched by pulseeffects. I am maintainer of pulseeffects in Fedora so I can apply any changes require for that.

**Comment by roberttucker** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1616070

**Comment by Vascom** [comment]
> https://bugzilla.redhat.com/show_bug.cgi?id=1616070

It is not enough.

**Comment by roberttucker** [comment]
I installed gstreamer1-plugins-bad-free-1.16.0-2.fc30 and gstreamer1-plugins-bad-free-devel using packages downloaded from here:
https://kojipkgs.fedoraproject.org/packages/gstreamer1-plugins-bad-free/1.16.0/2.fc30/x86_64/
I now have more plugins working, including "equalizer", which I use most. No "reverb" my second favourite.

**Comment by Vascom** [comment]
So why I have no Equalizer?

**Comment by roberttucker** [comment]
Did you install the update (yesterday's) from the Koji reference I put above? It needs to be this latest one.

**Comment by Vascom** [comment]
Yes.

**Comment by roberttucker** [comment]
And you've set it as installed in DConf – you can adjust the equalizer settings, it just doesn't produce an output, nor show an input?

**Comment by Vascom** [comment]
I use KDE and have no dconf.
Pulseeffects not can't find this plugin at all.

**Comment by roberttucker** [comment]
What do you get with:
$ gst-inspect-1.0 | grep -i lsp

**Comment by roberttucker** [comment]
In DConf (GUI)go to /com/github/wwmm/pulseeffects/sinkinputs and /com/github/wwmm/pulseeffects/sourceoutputs and see that all the ones you want are installed (moving the slider if necessary).

**Comment by Vascom** [comment]
gst-inspect-1.0 | grep -i lsp
give nothing.

**Comment by wwmm** [comment]
Using dconf to turn on plugins is not necessary unless your installation has some weird GSettings bug preventing PE of doing this itself.

`gst-plugin-scanner` warnings can be ignored. GStreamer started to show them a few releases ago on Arch Linux too. In my opinion it is a bug in GStreamer. But it is harmless to PE.

As GStreamer cache bug seems to be back in some situations it may be worth to clear its cache again `rm -rf ~/.cache/gstreamer-1.0`. This may be the reason why you still have missing plugins.

**Comment by wwmm** [comment]
The `compressor`, `delay` and the `equalizer` plugins are from LSP plugins(LV2 libraries). You should have something like this in the output of  `gst-inspect-1.0 | grep -i lsp`:
```
lv2:  lsp-plug-in-plugins-lv2-para-equalizer-x32-lr: Parametric Equalizer x32 LeftRight - Parametrischer Entzerrer x32 LeftRight

lv2:  lsp-plug-in-plugins-lv2-compressor-stereo: Compressor Stereo - Kompressor Stereo

lv2:  lsp-plug-in-plugins-lv2-comp-delay-x2-stereo: Delay Compensator x2 Stereo - Verzögerungsausgleicher x2 Stereo
```
As Calf plugins are being found there a few possible causes. GStreamer cache may not have been updated. LSP may have installed VST libraries or LADSPA libraries but not LV2 libraries.

The `crossfeed` plugin is in a similar situation as `lilv`. `gst-plugins-bad` has to be compiled with `bs2b` library support. Once this is done you will see `bs2b:  bs2b: Crossfeed effect` in the output of `gst-inspect-1.0 | grep -i bs2b`. As Arch Linux has been compiling `gst-plugins-bad` with bs2b and lilv support for months I assumed that at this point all major distributions would be doing the same. It is sad to see this is not the case =/

**Comment by wwmm** [comment]
@roberttucker the `reverb` plugin is from Calf. If other plugins like the limiter, bass enhancer and the exciter are working the `reverb` should be too. This is the output of `gst-inspect-1.0 | grep -i calf` on Arch Linux
```
lv2:  calf-sourceforge-net-plugins-Analyzer: Calf Analyzer
lv2:  calf-sourceforge-net-plugins-BassEnhancer: Calf Bass Enhancer
lv2:  calf-sourceforge-net-plugins-CompensationDelay: Calf Compensation Delay Line
lv2:  calf-sourceforge-net-plugins-Compressor: Calf Compressor
lv2:  calf-sourceforge-net-plugins-Crusher: Calf Crusher
lv2:  calf-sourceforge-net-plugins-Deesser: Calf Deesser
lv2:  calf-sourceforge-net-plugins-Emphasis: Calf Emphasis
lv2:  calf-sourceforge-net-plugins-Equalizer12Band: Calf Equalizer 12 Band
lv2:  calf-sourceforge-net-plugins-Equalizer30Band: Calf Equalizer 30 Band
lv2:  calf-sourceforge-net-plugins-Equalizer5Band: Calf Equalizer 5 Band
lv2:  calf-sourceforge-net-plugins-Equalizer8Band: Calf Equalizer 8 Band
lv2:  calf-sourceforge-net-plugins-Exciter: Calf Exciter
lv2:  calf-sourceforge-net-plugins-Filter: Calf Filter
lv2:  calf-sourceforge-net-plugins-Filterclavier: Calf Filterclavier
lv2:  calf-sourceforge-net-plugins-Flanger: Calf Flanger
lv2:  calf-sourceforge-net-plugins-Fluidsynth: Calf Fluidsynth
lv2:  calf-sourceforge-net-plugins-Gate: Calf Gate
lv2:  calf-sourceforge-net-plugins-HaasEnhancer: Calf Haas Stereo Enhancer
lv2:  calf-sourceforge-net-plugins-Limiter: Calf Limiter
lv2:  calf-sourceforge-net-plugins-MonoCompressor: Calf Mono Compressor
lv2:  calf-sourceforge-net-plugins-MonoInput: Calf Mono Input
lv2:  calf-sourceforge-net-plugins-Monosynth: Calf Monosynth
lv2:  calf-sourceforge-net-plugins-MultiChorus: Calf Multi Chorus
lv2:  calf-sourceforge-net-plugins-MultiSpread: Calf Multi Spread
lv2:  calf-sourceforge-net-plugins-MultibandCompressor: Calf Multiband Compressor
lv2:  calf-sourceforge-net-plugins-MultibandEnhancer: Calf Multiband Enhancer
lv2:  calf-sourceforge-net-plugins-MultibandGate: Calf Multiband Gate
lv2:  calf-sourceforge-net-plugins-MultibandLimiter: Calf Multiband Limiter
lv2:  calf-sourceforge-net-plugins-Organ: Calf Organ
lv2:  calf-sourceforge-net-plugins-Phaser: Calf Phaser
lv2:  calf-sourceforge-net-plugins-Pitch: Calf Pitch Tools
lv2:  calf-sourceforge-net-plugins-Pulsator: Calf Pulsator
lv2:  calf-sourceforge-net-plugins-Reverb: Calf Reverb
lv2:  calf-sourceforge-net-plugins-ReverseDelay: Calf Reverse Delay
lv2:  calf-sourceforge-net-plugins-RingModulator: Calf Ring Modulator
lv2:  calf-sourceforge-net-plugins-RotarySpeaker: Calf Rotary Speaker
lv2:  calf-sourceforge-net-plugins-Saturator: Calf Saturator
lv2:  calf-sourceforge-net-plugins-StereoTools: Calf Stereo Tools
lv2:  calf-sourceforge-net-plugins-TapeSimulator: Calf Tape Simulator
lv2:  calf-sourceforge-net-plugins-TransientDesigner: Calf Transient Designer
lv2:  calf-sourceforge-net-plugins-VintageDelay: Calf Vintage Delay
lv2:  calf-sourceforge-net-plugins-Vinyl: Calf Vinyl
lv2:  calf-sourceforge-net-plugins-Wavetable: Calf Wavetable
lv2:  calf-sourceforge-net-plugins-XOver2Band: Calf X-Over 2 Band
lv2:  calf-sourceforge-net-plugins-XOver3Band: Calf X-Over 3 Band
lv2:  calf-sourceforge-net-plugins-XOver4Band: Calf X-Over 4 Band
```
Removing GStreamer cache has no effect?

**Comment by Vascom** [comment]
Thanks. I will try it all tomorrow.

**Comment by roberttucker** [comment]
I didn't have lv2-calf-plugins installed, so installed it from repo, and now the only ones I haven't got working are: Convolver, Crystalizer, Crossfeed, Loudness, Maximizer, and Pitch. The only one of these I used is Crystalizer, which (oddly?) did work before installing the LSP plugins.

**Comment by wwmm** [comment]
This is odd. `Convolver` and `crystalizer` come with PE sources. Unless they were split in a secondary package like it is done in Ubuntu they should have been installed with PE. The command `gst-inspect-1.0 | grep -i pulseeffects` should return this
```
pecrystalizer:  pecrystalizer: PulseEffects Crystalizer
peconvolver:  peconvolver: PulseEffects Convolver
peautogain:  peautogain: PulseEffects ebur128 level meter
```

The `crossfeed` has the requirements I told in a post above. The `maximizer` needs ZamAudio plugins http://www.zamaudio.com/?p=976. The `loudness` plugins needs MDA.LV2 https://drobilla.net/software/mda-lv2. And the `pitch` plugin needs `rubberband` LADSPA library.

**Comment by roberttucker** [comment]
$ gst-inspect-1.0 | grep -i pulseeffects
peautogain:  peautogain: PulseEffects ebur128 level meter

My current installation of PE was built from source.

**Comment by wwmm** [comment]
What meson prints when it is configuring the build folder?

**Comment by wwmm** [comment]
This is what I have on Arch Linux
```
wallace@wwmm ~/myfiles/code/cpp/pulseeffects $ meson _build --prefix=/tmp/pe
The Meson build system
Version: 0.51.0
Source dir: /home/wallace/myfiles/code/cpp/pulseeffects
Build dir: /home/wallace/myfiles/code/cpp/pulseeffects/_build
Build type: native build
Project name: pulseeffects
Project version: 4.6.5
C compiler for the build machine: ccache cc (gcc 9.1.0 "cc (GCC) 9.1.0")
C++ compiler for the build machine: ccache c++ (gcc 9.1.0 "c++ (GCC) 9.1.0")
C compiler for the host machine: ccache cc (gcc 9.1.0 "cc (GCC) 9.1.0")
C++ compiler for the host machine: ccache c++ (gcc 9.1.0 "c++ (GCC) 9.1.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Configuring config.h using configuration
Program appstream-util found: YES (/usr/bin/appstream-util)
Program desktop-file-validate found: YES (/usr/bin/desktop-file-validate)
Configuring about.glade using configuration
Found pkg-config: /usr/bin/pkg-config (1.6.1)
Configuring com.github.wwmm.pulseeffects.service using configuration
Program itstool found: YES (/usr/bin/itstool)
Found pkg-config: /usr/bin/pkg-config (1.6.1)
Run-time dependency libpulse found: YES 12.2
Run-time dependency glib-2.0 found: YES 2.60.4
Run-time dependency glibmm-2.4 found: YES 2.60.0
Run-time dependency gtk+-3.0 found: YES 3.24.9
Run-time dependency gtkmm-3.0 found: YES 3.24.1
Run-time dependency sigc++-2.0 found: YES 2.10.2
Run-time dependency gstreamer-1.0 found: YES 1.16.0
Run-time dependency gstreamer-plugins-bad-1.0 found: YES 1.16.0
Run-time dependency gstreamer-fft-1.0 found: YES 1.16.0
Run-time dependency lilv-0 found: YES 0.24.4
Run-time dependency libbs2b found: YES 3.1.0
Run-time dependency Boost (system, filesystem) found: YES 1.69
Run-time dependency sndfile found: YES 1.0.28
Run-time dependency threads found: YES 
Library zita-convolver found: YES
Dependency gstreamer-1.0 found: YES (cached)
Run-time dependency gstreamer-base-1.0 found: YES 1.16.0
Run-time dependency gstreamer-controller-1.0 found: YES 1.16.0
Run-time dependency gstreamer-audio-1.0 found: YES 1.16.0
Dependency sndfile found: YES (cached)
Run-time dependency samplerate found: YES 0.1.9
Dependency threads found: YES (cached)
Library zita-convolver found: YES
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES (cached)
Dependency gstreamer-controller-1.0 found: YES (cached)
Dependency gstreamer-audio-1.0 found: YES (cached)
Run-time dependency libebur128 found: YES 1.2.4
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES (cached)
Dependency gstreamer-controller-1.0 found: YES (cached)
Dependency gstreamer-audio-1.0 found: YES (cached)
Dependency libebur128 found: YES (cached)
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES (cached)
Dependency gstreamer-controller-1.0 found: YES (cached)
Dependency gstreamer-audio-1.0 found: YES (cached)
Program meson_post_install.py found: YES (/usr/bin/python /home/wallace/myfiles/code/cpp/pulseeffects/meson_post_install.py)
Build targets in project: 14
Found ninja-1.9.0 at /usr/bin/ninja
```

**Comment by roberttucker** [comment]
Mine on Fedora 30:
`$ meson _build --prefix=/usr
The Meson build system
Version: 0.50.1
Source dir: /home/rt/Programs/pulseeffects
Build dir: /home/rt/Programs/pulseeffects/_build
Build type: native build
Project name: pulseeffects
Project version: 4.6.5
Native C compiler: cc (gcc 9.1.1 "cc (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
Native C++ compiler: c++ (gcc 9.1.1 "c++ (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Configuring config.h using configuration
Program appstream-util found: YES (/usr/bin/appstream-util)
Program desktop-file-validate found: YES (/usr/bin/desktop-file-validate)
Configuring about.glade using configuration
Found pkg-config: /usr/bin/pkg-config (1.6.1)
Configuring com.github.wwmm.pulseeffects.service using configuration
Program itstool found: YES (/usr/bin/itstool)
Dependency libpulse found: YES 12.2-rebootstrapped
Dependency glib-2.0 found: YES 2.60.4
Dependency glibmm-2.4 found: YES 2.60.0
Dependency gtk+-3.0 found: YES 3.24.9
Dependency gtkmm-3.0 found: YES 3.24.1
Dependency sigc++-2.0 found: YES 2.10.2
Dependency gstreamer-1.0 found: YES 1.16.0
Dependency gstreamer-plugins-bad-1.0 found: YES 1.16.0
Dependency gstreamer-fft-1.0 found: YES 1.16.0
Dependency lilv-0 found: YES 0.24.4
Dependency libbs2b found: YES 3.1.0
Dependency Boost (system, filesystem) found: YES 1.69
Dependency sndfile found: YES 1.0.28
Dependency threads found: YES 
Library zita-convolver found: YES
Message: Missing dependency zita-convolver = 3.x.x or zita-convolver = 4.x.x
Message: Convolver plugin will not be built
Library zita-convolver found: YES
Message: Missing dependency zita-convolver = 3.x.x or zita-convolver = 4.x.x
Message: Convolver plugin will not be built
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES 1.16.0
Dependency gstreamer-controller-1.0 found: YES 1.16.0
Dependency gstreamer-audio-1.0 found: YES 1.16.0
Dependency libebur128 found: YES 1.2.3
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES (cached)
Dependency gstreamer-controller-1.0 found: YES (cached)
Dependency gstreamer-audio-1.0 found: YES (cached)
Program meson_post_install.py found: YES (/usr/bin/env python3 /home/rt/Programs/pulseeffects/meson_post_install.py)
Build targets in project: 12
Found ninja-1.9.0 at /usr/bin/ninja
`

**Comment by wwmm** [comment]
That is the problem
```
Message: Missing dependency zita-convolver = 3.x.x or zita-convolver = 4.x.x
Message: Convolver plugin will not be built
```

**Comment by wwmm** [comment]
You must have zita 2.x

**Comment by wwmm** [comment]
And now I see that crystalizer `meson.build` is printing `Convolver` instead of `Crystalizer` when the wrong zita version is found. I will fix this. In case you need zita 3.x or 4.x.

**Comment by roberttucker** [comment]
No: zita-convolver-3.1.0-17.fc30.x86_64
Odd that it prints it out twice and at first says it has found it?

**Comment by wwmm** [comment]
This is how the check is being done in meson 
```
zita_convolver = cxx.find_library('zita-convolver', required: false)

if cxx.compiles(
'''
#include <zita-convolver.h>
#if ZITA_CONVOLVER_MAJOR_VERSION != 3 && ZITA_CONVOLVER_MAJOR_VERSION != 4
    #error "This programs requires zita-convolver 3 or 4"
#endif
''', dependencies: [zita_convolver])
```
The first yes is because  the library was found. But maybe you do not have the headers. This would make the second test fail.

**Comment by roberttucker** [comment]
I have zita-convolver-devel-3.1.0-17.fc30.x86_64 installed as well.

**Comment by wwmm** [comment]
Then I have no idea of what is happening. On Arch Linux these are the contents of the `zita-convolver 4.0.3` package
```
zita-convolver /usr/
zita-convolver /usr/include/
zita-convolver /usr/include/zita-convolver.h
zita-convolver /usr/lib/
zita-convolver /usr/lib/libzita-convolver.so
zita-convolver /usr/lib/libzita-convolver.so.4
zita-convolver /usr/lib/libzita-convolver.so.4.0.3
zita-convolver /usr/share/
zita-convolver /usr/share/doc/
zita-convolver /usr/share/doc/zita-convolver/
zita-convolver /usr/share/doc/zita-convolver/AUTHORS
zita-convolver /usr/share/doc/zita-convolver/README
```

**Comment by roberttucker** [comment]
`$ sudo find / -name *zita*
find: ‘/run/user/1000/gvfs’: Permission denied
/usr/lib64/libzita-convolver.so.3
/usr/lib64/libzita-convolver.so.3.1.0
/usr/lib64/libzita-convolver.so
/usr/include/zita-convolver.h
/usr/share/licenses/zita-convolver
/usr/share/doc/zita-convolver`

**Comment by wwmm** [comment]
When I first wrote the convolver plugin Arch Linux used zita 3 too. At least at that time zita 3 was found with the meson check above. I don't understand why this is not happening in Fedora too.

**Comment by Vascom** [comment]
May be I should open new issue?

**Comment by roberttucker** [comment]
Vasiliy, from what I can see from what you've put above you need to go to:
https://sourceforge.net/projects/lsp-plugins/files/lsp-plugins/1.1.9/Linux-x86_64/
and download and install the lv2 plugins.
Extract the tarball and copy the folder (with contents – I used Krusader):
lsp-plugins.lv2 --> /usr/lib64/lv2
If the ladspa plugins are needed, then untar that tarball and just copy the lsp-plugins-ladspa.so file to /usr/lib64/ladspa.

**Comment by Vascom** [comment]
I already have lv2 plugins installed from package:
```
$ gst-inspect-1.0 | grep -i lv2
lv2:  drobilla-net-plugins-mda-Vocoder: MDA Vocoder
lv2:  drobilla-net-plugins-mda-VocInput: MDA VocInput
lv2:  drobilla-net-plugins-mda-Transient: MDA Transient
lv2:  drobilla-net-plugins-mda-Tracker: MDA Tracker
lv2:  drobilla-net-plugins-mda-ThruZero: MDA ThruZero
lv2:  drobilla-net-plugins-mda-TestTone: MDA TestTone
lv2:  drobilla-net-plugins-mda-TalkBox: MDA TalkBox
lv2:  drobilla-net-plugins-mda-SubSynth: MDA SubSynth
lv2:  drobilla-net-plugins-mda-Stereo: MDA Stereo
lv2:  drobilla-net-plugins-mda-Splitter: MDA Splitter
lv2:  drobilla-net-plugins-mda-Shepard: MDA Shepard
lv2:  drobilla-net-plugins-mda-RoundPan: MDA RoundPan
lv2:  drobilla-net-plugins-mda-RingMod: MDA RingMod
lv2:  drobilla-net-plugins-mda-RezFilter: MDA RezFilter
lv2:  drobilla-net-plugins-mda-RePsycho: MDA RePsycho!
lv2:  drobilla-net-plugins-mda-Piano: MDA Piano
lv2:  drobilla-net-plugins-mda-Overdrive: MDA Overdrive
lv2:  drobilla-net-plugins-mda-MultiBand: MDA MultiBand
lv2:  drobilla-net-plugins-mda-Loudness: MDA Loudness
lv2:  drobilla-net-plugins-mda-Limiter: MDA Limiter
lv2:  drobilla-net-plugins-mda-Leslie: MDA Leslie
lv2:  drobilla-net-plugins-mda-JX10: MDA JX10
lv2:  drobilla-net-plugins-mda-Image: MDA Image
lv2:  drobilla-net-plugins-mda-EPiano: MDA ePiano
lv2:  drobilla-net-plugins-mda-Dynamics: MDA Dynamics
lv2:  drobilla-net-plugins-mda-DubDelay: MDA DubDelay
lv2:  drobilla-net-plugins-mda-Dither: MDA Dither
lv2:  drobilla-net-plugins-mda-Detune: MDA Detune
lv2:  drobilla-net-plugins-mda-Delay: MDA Delay
lv2:  drobilla-net-plugins-mda-Degrade: MDA Degrade
lv2:  drobilla-net-plugins-mda-DeEss: MDA De-ess
lv2:  drobilla-net-plugins-mda-DX10: MDA DX10
lv2:  drobilla-net-plugins-mda-Combo: MDA Combo
lv2:  drobilla-net-plugins-mda-BeatBox: MDA BeatBox
lv2:  drobilla-net-plugins-mda-Bandisto: MDA Bandisto
lv2:  drobilla-net-plugins-mda-Ambience: MDA Ambience
lv2:  calf-sourceforge-net-plugins-XOver4Band: Calf X-Over 4 Band
lv2:  calf-sourceforge-net-plugins-XOver3Band: Calf X-Over 3 Band
lv2:  calf-sourceforge-net-plugins-XOver2Band: Calf X-Over 2 Band
lv2:  calf-sourceforge-net-plugins-Wavetable: Calf Wavetable
lv2:  calf-sourceforge-net-plugins-Vinyl: Calf Vinyl
lv2:  calf-sourceforge-net-plugins-VintageDelay: Calf Vintage Delay
lv2:  calf-sourceforge-net-plugins-TransientDesigner: Calf Transient Designer
lv2:  calf-sourceforge-net-plugins-TapeSimulator: Calf Tape Simulator
lv2:  calf-sourceforge-net-plugins-StereoTools: Calf Stereo Tools
lv2:  calf-sourceforge-net-plugins-Saturator: Calf Saturator
lv2:  calf-sourceforge-net-plugins-RotarySpeaker: Calf Rotary Speaker
lv2:  calf-sourceforge-net-plugins-RingModulator: Calf Ring Modulator
lv2:  calf-sourceforge-net-plugins-ReverseDelay: Calf Reverse Delay
lv2:  calf-sourceforge-net-plugins-Reverb: Calf Reverb
lv2:  calf-sourceforge-net-plugins-Pulsator: Calf Pulsator
lv2:  calf-sourceforge-net-plugins-Pitch: Calf Pitch Tools
lv2:  calf-sourceforge-net-plugins-Phaser: Calf Phaser
lv2:  calf-sourceforge-net-plugins-Organ: Calf Organ
lv2:  calf-sourceforge-net-plugins-MultibandLimiter: Calf Multiband Limiter
lv2:  calf-sourceforge-net-plugins-MultibandGate: Calf Multiband Gate
lv2:  calf-sourceforge-net-plugins-MultibandEnhancer: Calf Multiband Enhancer
lv2:  calf-sourceforge-net-plugins-MultibandCompressor: Calf Multiband Compressor
lv2:  calf-sourceforge-net-plugins-MultiSpread: Calf Multi Spread
lv2:  calf-sourceforge-net-plugins-MultiChorus: Calf Multi Chorus
lv2:  calf-sourceforge-net-plugins-Monosynth: Calf Monosynth
lv2:  calf-sourceforge-net-plugins-MonoInput: Calf Mono Input
lv2:  calf-sourceforge-net-plugins-MonoCompressor: Calf Mono Compressor
lv2:  calf-sourceforge-net-plugins-Limiter: Calf Limiter
lv2:  calf-sourceforge-net-plugins-HaasEnhancer: Calf Haas Stereo Enhancer
lv2:  calf-sourceforge-net-plugins-Gate: Calf Gate
lv2:  calf-sourceforge-net-plugins-Fluidsynth: Calf Fluidsynth
lv2:  calf-sourceforge-net-plugins-Flanger: Calf Flanger
lv2:  calf-sourceforge-net-plugins-Filterclavier: Calf Filterclavier
lv2:  calf-sourceforge-net-plugins-Filter: Calf Filter
lv2:  calf-sourceforge-net-plugins-Exciter: Calf Exciter
lv2:  calf-sourceforge-net-plugins-Equalizer8Band: Calf Equalizer 8 Band
lv2:  calf-sourceforge-net-plugins-Equalizer5Band: Calf Equalizer 5 Band
lv2:  calf-sourceforge-net-plugins-Equalizer30Band: Calf Equalizer 30 Band
lv2:  calf-sourceforge-net-plugins-Equalizer12Band: Calf Equalizer 12 Band
lv2:  calf-sourceforge-net-plugins-Emphasis: Calf Emphasis
lv2:  calf-sourceforge-net-plugins-Deesser: Calf Deesser
lv2:  calf-sourceforge-net-plugins-Crusher: Calf Crusher
lv2:  calf-sourceforge-net-plugins-Compressor: Calf Compressor
lv2:  calf-sourceforge-net-plugins-CompensationDelay: Calf Compensation Delay Line
lv2:  calf-sourceforge-net-plugins-BassEnhancer: Calf Bass Enhancer
lv2:  calf-sourceforge-net-plugins-Analyzer: Calf Analyzer
```

**Comment by roberttucker** [comment]
You need the LSP ones for PE.

**Comment by Vascom** [comment]
Thanks!
Now it work. I will add this plugins to Fedora repo.

**Comment by roberttucker** [comment]
Looks as if the zita-convolver package might do with an update/upgrade too.

**Comment by Vascom** [comment]
zita-convolver updated now in rawhide.

**Comment by roberttucker** [comment]
Not zita-convolver-4?

**Comment by Vascom** [comment]
4.0.3
https://koji.fedoraproject.org/koji/packageinfo?packageID=8438

**Comment by roberttucker** [comment]
Is there no devl package?

**Comment by Vascom** [comment]
It exist https://koji.fedoraproject.org/koji/buildinfo?buildID=1297084

**Comment by roberttucker** [comment]
Thanks.
`$ sudo dnf install zita-convolver zita-convolver-devel
Last metadata expiration check: 0:00:24 ago on Wed 26 Jun 2019 09:22:02 BST.
Package zita-convolver-4.0.3-2.fc31.x86_64 is already installed.
Package zita-convolver-devel-4.0.3-2.fc31.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!`
Unfortunately PE is still not finding it:
`$ meson _build --prefix=/usr
The Meson build system
Version: 0.50.1
Source dir: /home/rt/Programs/pulseeffects
Build dir: /home/rt/Programs/pulseeffects/_build
Build type: native build
Project name: pulseeffects
Project version: 4.6.5
Native C compiler: cc (gcc 9.1.1 "cc (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
Native C++ compiler: c++ (gcc 9.1.1 "c++ (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Configuring config.h using configuration
Program appstream-util found: YES (/usr/bin/appstream-util)
Program desktop-file-validate found: YES (/usr/bin/desktop-file-validate)
Configuring about.glade using configuration
Found pkg-config: /usr/bin/pkg-config (1.6.1)
Configuring com.github.wwmm.pulseeffects.service using configuration
Program itstool found: YES (/usr/bin/itstool)
Dependency libpulse found: YES 12.2-rebootstrapped
Dependency glib-2.0 found: YES 2.60.4
Dependency glibmm-2.4 found: YES 2.60.0
Dependency gtk+-3.0 found: YES 3.24.9
Dependency gtkmm-3.0 found: YES 3.24.1
Dependency sigc++-2.0 found: YES 2.10.2
Dependency gstreamer-1.0 found: YES 1.16.0
Dependency gstreamer-plugins-bad-1.0 found: YES 1.16.0
Dependency gstreamer-fft-1.0 found: YES 1.16.0
Dependency lilv-0 found: YES 0.24.4
Dependency libbs2b found: YES 3.1.0
Dependency Boost (system, filesystem) found: YES 1.69
Dependency sndfile found: YES 1.0.28
Dependency threads found: YES 
Library zita-convolver found: YES
Message: Missing dependency zita-convolver = 3.x.x or zita-convolver = 4.x.x
Message: Convolver plugin will not be built
Library zita-convolver found: YES
Message: Missing dependency zita-convolver = 3.x.x or zita-convolver = 4.x.x
Message: Crystalizer plugin will not be built
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES 1.16.0
Dependency gstreamer-controller-1.0 found: YES 1.16.0
Dependency gstreamer-audio-1.0 found: YES 1.16.0
Dependency libebur128 found: YES 1.2.3
Dependency gstreamer-1.0 found: YES (cached)
Dependency gstreamer-base-1.0 found: YES (cached)
Dependency gstreamer-controller-1.0 found: YES (cached)
Dependency gstreamer-audio-1.0 found: YES (cached)
Program meson_post_install.py found: YES (/usr/bin/env python3 /home/rt/Programs/pulseeffects/meson_post_install.py)
Build targets in project: 12
Found ninja-1.9.0 at /usr/bin/ninja`

**Comment by Vascom** [comment]
Package from repo can find it
https://kojipkgs.fedoraproject.org//packages/pulseeffects/4.6.5/2.fc31/data/logs/x86_64/build.log

**Comment by roberttucker** [comment]
OK, great, thanks, all working now!

**Comment by wwmm** [comment]
I think we can close this.

**Comment by roberttucker** [comment]
Seem to be hitting the same problem again with zita-convolver.
Both pulseeffects-4.6.6-1.fc31.x86_64.rpm
 and source code compilation with zita-convolver and zita-convolver-devel 4.0.3-3.fc32.x86_64

**Comment by roberttucker** [comment]
Solved:
Seems I needed to install fftw (and fftw-devel?).

**Comment by wwmm** [comment]
If you are compiling zita yourself you probably will need `fftw-devel` too. On Arch Linux the zita package depends on `fftw` but on Arch Linux this package offers both libraries and headers.

**Comment by roberttucker** [comment]
The meson log just said fftw.h could not be found, so I just installed both.

**Comment by Vascom** [comment]
pulseeffects-4.6.6-1.fc31.x86_64.rpm already requires libzita-convolver.so.4()(64bit) and it requires libfftw3f.so.3()(64bit). So fftw must be installed automatically.
If you see any problem with rpm package please report it to [bugzilla](https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&version=rawhide&component=pulseeffects)
